From: Russ Allbery <rra@cpan.org>
Date: Tue, 22 Apr 2014 16:49:53 -0700
Subject: Avoid using an undefined value in check_funcs

When the caller didn't pass in a notfound function, we were still
constructing one with an undefined value.  Stop doing that.
---
 Build.PL | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Build.PL b/Build.PL
index 57f54f2..ccbdfc3 100644
--- a/Build.PL
+++ b/Build.PL
@@ -115,7 +115,10 @@ sub check_funcs {
     my $return = 1;
     for my $function (@{$functions_ref}) {
         my $found_ref    = sub { $func_found_ref->($function) };
-        my $notfound_ref = sub { $user_notfound_ref->($function) };
+        my $notfound_ref;
+        if (defined($user_notfound_ref)) {
+            $notfound_ref = sub { $user_notfound_ref->($function) };
+        }
         $return &= check_func($self, $function, $found_ref, $notfound_ref);
     }
     return $return;
